contracts/ExoticPositionalMarkets/ThalesBonds.sol

```diff
@@ -0,0 +1,284 @@
+        setOwner(_owner);
+        initNonReentrant();
+    }
+
+    function getTotalDepositedBondAmountForMarket(address _market) external view returns (uint) {
+        return marketBond[_market].totalDepositedMarketBond;
+    }
+
+    function getClaimedBondAmountForMarket(address _market) external view returns (uint) {
+        return marketBond[_market].totalDepositedMarketBond.sub(marketBond[_market].totalMarketBond);
+    }
+
+    function getClaimableBondAmountForMarket(address _market) external view returns (uint) {
+        return marketBond[_market].totalMarketBond;
+    }
+
+    function getDisputorBondForMarket(address _market, address _disputorAddress) external view returns (uint) {
+        return marketBond[_market].disputorBond[_disputorAddress];
+    }
+
+    function getCreatorBondForMarket(address _market) external view returns (uint) {
+        return marketBond[_market].creatorBond;
+    }
+
+    function getResolverBondForMarket(address _market) external view returns (uint) {
+        return marketBond[_market].resolverBond;
+    }
+
+    // different deposit functions to flag the bond amount : creator
+    function sendCreatorBondToMarket(
+        address _market,
+        address _creatorAddress,
+        uint _amount
+    ) external onlyOracleCouncilManagerAndOwner nonReentrant {
+        require(_amount > 0, "Bond zero");
+        require(_market != address(0), "Invalid address");
+        marketBond[_market].creatorBond = _amount;
+        marketBond[_market].totalMarketBond = marketBond[_market].totalMarketBond.add(_amount);
+        marketBond[_market].totalDepositedMarketBond = marketBond[_market].totalDepositedMarketBond.add(_amount);
+        transferToMarketBond(_creatorAddress, _amount);
+        emit CreatorBondSent(_market, _creatorAddress, _amount);
+    }
+
+    // different deposit functions to flag the bond amount : resolver
+    function sendResolverBondToMarket(
+        address _market,
+        address _resolverAddress,
+        uint _amount
+    ) external onlyOracleCouncilManagerAndOwner nonReentrant {
+        require(_amount > 0, "Bond zero");
+        require(_market != address(0), "Invalid address");
+        // in case the creator is the resolver, move the bond to the resolver
+        marketBond[_market].resolverBond = _amount;
+        marketBond[_market].totalMarketBond = marketBond[_market].totalMarketBond.add(_amount);
+        marketBond[_market].totalDepositedMarketBond = marketBond[_market].totalDepositedMarketBond.add(_amount);
+        transferToMarketBond(_resolverAddress, _amount);
+        emit ResolverBondSent(_market, _resolverAddress, _amount);
+    }
+
+    // different deposit functions to flag the bond amount : disputor
+    function sendDisputorBondToMarket(
+        address _market,
+        address _disputorAddress,
+        uint _amount
+    ) external onlyOracleCouncilManagerAndOwner nonReentrant {
+        require(_amount > 0, "Bond zero");
```

Comment:
doesnt hurt to add a check if this is a known market in the MarketManager (applies to all method that consume a market as a parameter)